components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2827,6 +2831,38 @@ public API createNewAPIVersion(String existingApiId, String newVersion, Boolean
         return getAPIbyUUID(newAPIId, organization);
     }
 
+    private void attachOperationPoliciesToAPI(API newAPI, Map<String, String> clonedOperationPolicyMap,
+                                              Map<String, List<OperationPolicy>> operationPoliciesMap)
+            throws APIManagementException {
+        operationPoliciesMap.forEach((key, operationPolicies) ->
+                operationPolicies.forEach(operationPolicy ->
+                        operationPolicy.setPolicyId(clonedOperationPolicyMap.get(operationPolicy.getPolicyId()))));
+        Set<URITemplate> uriTemplates = newAPI.getUriTemplates();
+        if (uriTemplates != null) {
+            for (URITemplate uriTemplate : uriTemplates) {
+                List<OperationPolicy> operationPolicies =
+                        operationPoliciesMap.get(uriTemplate.getHTTPVerb() + ":" + uriTemplate.getUriTemplate());
+                uriTemplate.setOperationPolicies(operationPolicies);
+            }
+            apiMgtDAO.addOperationPolicyMapping(uriTemplates);
+        }
+    }
+
+    private Map<String, String> cloneOperationPoliciesToAPI(API newAPI, Map<String, List<OperationPolicy>> operationPoliciesMap)
+            throws APIManagementException {
+        Map<String, String> clonedPolicies = new HashMap<>();
+        for (Map.Entry<String, List<OperationPolicy>> operationPolicyEntry : operationPoliciesMap.entrySet()) {
+            List<OperationPolicy> operationPolicyList = operationPolicyEntry.getValue();
+            for (OperationPolicy operationPolicy : operationPolicyList) {
+                if (clonedPolicies.containsKey(operationPolicy.getPolicyId())) {
+                    String policyUUID = apiMgtDAO.cloneOperationPolicy(operationPolicy.getPolicyId(), newAPI.getUuid(), null);
```

Comment:
Can't we straight away set the new policyId to the policy at here? This way we can reduce the step at https://github.com/wso2/carbon-apimgt/pull/11050/files#diff-19a032cb823fd4f31e79376db6427e305b79949a26979f42d5e7ea0c607cdff2R2837. Same API object is sent over to this method?